ui/app/mirrors/create/cdc/schemabox.tsx

```diff
@@ -93,47 +113,37 @@ const SchemaBox = ({
       ?.columns;
   };
 
-  const handleColumnExclusion = (
-    source: string,
-    column: string,
-    include: boolean
-  ) => {
-    const currRows = [...rows];
-    const rowOfSource = currRows.find((row) => row.source === source);
-    if (rowOfSource) {
-      if (include) {
-        const updatedExclude = rowOfSource.exclude.filter(
-          (col) => col !== column
-        );
-        rowOfSource.exclude = updatedExclude;
-      } else {
-        rowOfSource.exclude.push(column);
-      }
-    }
-    setRows(currRows);
-  };
-
   const handleSelectAll = (
-    e: React.MouseEvent<HTMLInputElement, MouseEvent>
+    e: React.MouseEvent<HTMLInputElement, MouseEvent>,
+    schemaName: string
   ) => {
+    setHandlingAll(true);
     const newRows = [...rows];
     for (const row of newRows) {
-      row.selected = e.currentTarget.checked;
-      if (e.currentTarget.checked) addTableColumns(row.source);
-      else removeTableColumns(row.source);
+      if (row.schema === schemaName) {
+        row.selected = e.currentTarget.checked;
+        if (e.currentTarget.checked) addTableColumns(row.source);
+        else removeTableColumns(row.source);
+      }
     }
     setRows(newRows);
+    setHandlingAll(false);
   };
 
   const handleSchemaClick = (schemaName: string) => {
     if (!schemaIsExpanded(schemaName)) {
       setTablesLoading(true);
       setExpandedSchemas((curr) => [...curr, schemaName]);
-      fetchTables(sourcePeer, schemaName, peerType).then((tableRows) => {
-        const newRows = [...rows, ...tableRows];
-        setRows(newRows);
-        setTablesLoading(false);
-      });
+      if (!schemaLoadedSet.has(schemaName)) {
+        const updatedSet = new Set(schemaLoadedSet);
+        updatedSet.add(schemaName); // Add the new schemaName
+        setSchemaLoadedSet(updatedSet);
```

Comment:
```suggestion
        setSchemaLoadedSet(loaded => new Set(loaded).add(schemaName));
```
Avoids race conditions